src/components/Paymethods.tsx

```diff
@@ -1,17 +1,40 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { TabContent, Nav, NavItem, NavLink } from "reactstrap";
 import { PaymentMethod } from "./PaymentMethod";
 import { useTranslation } from "react-i18next";
 import "./Paymethods.css";
 import { PaymentMethod as PaymentMethodDef } from "../../generated/definitions/PaymentMethod";
+import { Wallet as WalletDef } from "../../generated/definitions/Wallet";
+import { PublicWalletItem } from "../../generated/definitions/PublicWalletItem";
 import classNames from "classnames";
+import { PublicCreditCard } from "../../generated/definitions/PublicCreditCard";
 
 type PaymethodsProps = {
   paylist: ReadonlyArray<PaymentMethodDef>;
+  wallet?: WalletDef;
 };
 
 export const Paymethods: React.FunctionComponent<PaymethodsProps> = props => {
   const [activeTab, setActiveTab] = useState<number>(0);
+  const [walletInfo, setWalletInfo] = useState<
+    Record<string, PublicWalletItem> | undefined
+  >(undefined);
+
+  function renderPAN(walletItem: PublicWalletItem): string {
+    return PublicCreditCard.is(walletItem)
+      ? ` | *${walletItem.masked_pan}`
+      : `| ${walletItem.type}`;
+  }
+
+  useEffect(() => {
+    if (props.wallet) {
+      const walletHash = props.wallet?.data.reduce(
+        (obj, item) => ({ ...obj, [item.hpan]: item }),
+        {}
+      );
+      setWalletInfo(walletHash);
+    }
```

Comment:
We can simplify with
```suggestion
   setWalletInfo(
      props.wallet?.data.reduce(
        (obj, item) => ({ ...obj, [item.hpan]: item }),
        {} as Record<string, PublicWalletItem>
      )
    );
```